This is shiny app to help accessing coexpression data generated in "Exploring the noncanonical translatome using massively integrated coexpression analysis."

doi: https://doi.org/10.1101/2023.03.16.533058 



@article {Rich2023.03.16.533058,
	author = {April Rich and Omer Acar and Anne-Ruxandra Carvunis},
	title = {Exploring the noncanonical translatome using massively integrated coexpression analysis},
	elocation-id = {2023.03.16.533058},
	year = {2023},
	doi = {10.1101/2023.03.16.533058},
	publisher = {Cold Spring Harbor Laboratory},
	abstract = {Cells transcribe and translate thousands of noncanonical open reading frames (nORFs) whose impacts on cellular phenotypes are unknown. Here, we investigated nORF transcription, evolution, and potential cellular roles using a coexpression approach. We measured coexpression between \~{}6,000 nORFs and \~{}6000 canonical ORFs (cORFs) in the Saccharomyces cerevisiae genome by massively integrating thousands of RNA sequencing samples and developing a dedicated computational framework that accounts for low expression levels. Our findings reveal that almost all cORFs are strongly coexpressed with at least one nORF. However, almost half of nORFs are not strongly coexpressed with any cORFs and form entirely new transcription modules. Many nORFs recently evolved de novo in genomic regions that were non-coding in the Saccharomyces ancestor. Coexpression profiles suggest that half of de novo nORFs are functionally associated with conserved genes involved in cellular transport or homeostasis. Furthermore, we discovered that de novo ORFs located downstream of conserved genes leverage their neighbors{\textquoteright} transcripts resulting in high expression levels. Where a de novo nORF emerges could be just as important as its sequence for shaping how it can influence cellular phenotype. Our coexpression dataset serves as an unprecedented resource for unraveling how nORFs integrate into cellular networks, contribute to cellular phenotypes and evolve.Competing Interest StatementA.-R.C. is a member of the scientific advisory board for Flagship Labs 69, Inc.},
	URL = {https://www.biorxiv.org/content/early/2023/03/17/2023.03.16.533058},
	eprint = {https://www.biorxiv.org/content/early/2023/03/17/2023.03.16.533058.full.pdf},
	journal = {bioRxiv}
}
